### ISSUE TITLE
[BUG] Startup error when using enum constant at runtime

### ISSUE THREAD

**Issue by lorenzofratus** [cap-js/cds-typer#119]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Nature of Your Project

TypeScript

### Current Behavior

I'm getting the following error while starting the CAP application whenever I reference enums in the same way as it is explained in the [documentation](https://cap.cloud.sap/docs/tools/cds-typer#enums).

```bash
❗️ ERROR on server start: ❗️

 @cds-models/db/index.js:9:102 - error TS2339: Property 'enum' does not exist on type 'LinkedDefinition'.

9 module.exports.SampleEnum = Object.fromEntries(Object.entries(cds.model.definitions['db.SampleEnum'].enum).map(([k,v]) => [k,v.val??k]))
                                                                                                       ~~~~
```

### Expected Behavior

I should be able to reference enum values using the `const` that is generated by cds-typer without any error.

### Steps To Reproduce

1. Define the enum in a `.cds` file:
```cds
namespace db;

type SampleEnum : String(3) enum {
    a = 'AAA';
    b = 'BBB';
}
```
2. Import and reference the enum in a `.ts` file (there is no compiler error at this step):
```typescript
import { SampleEnum } from "#cds-models/db";
...
someVariable === SampleEnum.a;
```
3. Launch the application using:
```bash
cds-ts watch
```
4. The error appears

### Environment

```markdown
- **OS**: Linux distro of SAP BAS (5.15.114-gardenlinux-cloud-amd64)
- **Node**: 18.14.2
- **npm**: 9.5.0
- **cds-typer**: 0.12.0
- **cds**: 7.4.2
```

### Repository Containing a Minimal Reproducible Example

https://github.com/lorenzofratus/cds-typer-enum-bug

### Anything else?

Please let me know if I'm missing some configuration or if I can provide more information.

**Comment by daogrady** [comment]
Hi Lorenzo,

thank you for your report! Sorry for the confusion, this is a bug that is already fixed in `main` but has not yet been found its way into a release.
Feel free to check if using `main` fixes the problem with your model. I will leave this issue open for now until the corresponding fix is actually released.

Best,
Daniel

**Comment by daogrady** [comment]
Hi Lorenzo,

I have just released cds-typer 0.13.0 which includes the fix, so upgrading your dependency should fix the problem. Feel free to reopen this issue if the problem persists.

Best,
Daniel
